### ISSUE TITLE
stuck at [racon::Polisher::polish] with half of consensus windows generated

### ISSUE THREAD

**Issue by kfergy** [isovic/racon#58]
Hey there,

Long story short, I have a hybrid assembly build with the first two steps of DBG2OLC, but couldn't install blasr, POS. As suggested [here](https://www.biostars.org/p/212727/#212773), I turned to racon for consensus calling. I first used minimap2 to create a PAF (483 MB) from the backbone.fasta (222 MB) and the input PacBio.fasta (13.5 GB). Note, these are sequel reads so I'm using FASTA files. 

Now, I'm at the racon stage, using the following command:
`racon -t 8 "/usr/path/pbTricho.fasta" "/usr/path/TrichoHybrid-approx-mapping.paf" "/usr/path/backbone_raw.fasta"`

This works for the first 30 minutes,
> [racon::Polisher::initialize] loaded target sequences
> [racon::Polisher::initialize] loaded sequences
> [racon::Polisher::initialize] loaded batch of overlaps
> [racon::Polisher::initialize] aligned overlap 2060216/2060216
> [racon::Polisher::initialize] transformed data into windows

but then the program is hung up on the following section
> [racon::Polisher::polish] generated consensus for window 118876/467058

At this point, `top` indicates that racon is sleeping, still using computing power and memory, but going nowhere. I've tried altering the quality check with `-q 0` just to make sure that isn't an issue, but it still gets stuck. So my question is, how do I get it to continue - is this a FASTA issue, or something else going off?

thx

**Comment by rvaser** [comment]
Hello,
I guess one thread entered an infinite loop due to a never before seen bug (on window 118876). Can you please modify `src/window.cpp` as described bellow and check whether the program hungs up again plus send me the output? Alternatively, you can send me your whole data and tell me which version of Racon you are using.

Best regards,
Robert

src/window.cpp modification at line 67:
```cpp
    if (id_ != 118876) return false;
    for (uint32_t i = 0; i < sequences_.size(); ++i) {
        printf("%s\n%s\n%u %u\n",
            std::string(sequences_[i].first, sequences_[i].second).c_str(),
            std::string(qualities_[i].first, qualities_[i].second).c_str(),
            positions_[i].first, positions_[i].second);
    }
```

**Comment by kfergy** [comment]
I'm using racon v1.0.1. Some clarification - with each rerun, `racon` gets stuck at a different window in this part of the process. Here is may latest iteration 

> [racon::Polisher::initialize] loaded target sequences
> [racon::Polisher::initialize] loaded sequences
> [racon::Polisher::initialize] loaded batch of overlaps
> [racon::Polisher::initialize] aligned overlap 2060216/2060216
> [racon::Polisher::initialize] transformed data into windows
> [racon::Polisher::polish] generated consensus for window 6691/467058

With this in mind, does the change to `src/window.cpp` make sense? In the meantime, I'll also send you my files via ftp in case.


Thanks for the quick reply!

**Comment by rvaser** [comment]
Hmm that is quite odd. Please do send me your files and I'll check what is wrong locally.

**Comment by rvaser** [comment]
I have obtained the files and am running racon. Will get back to you hopefully soon.

**Comment by rvaser** [comment]
I found the bug! One of your reads had an insert of size 5kbp into a window of 500bp on the reference (maybe something is misassembled) which exceeded my expectation of maximal sequence length added to a window. Will fix it soon.

**Comment by rvaser** [comment]
Fixed in release 1.0.2!
